pkg/service/node.go

```diff
@@ -220,29 +221,43 @@ func (n *NodeService) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	}
 
 	targetPath := req.GetTargetPath()
-	notMnt, err := n.fsMounter.IsLikelyNotMountPoint(targetPath)
+	notMnt, err := n.mounter.IsLikelyNotMountPoint(targetPath)
 	if err != nil && !os.IsNotExist(err) {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 	if !notMnt {
 		klog.Infof("Volume %s already mounted on %s", req.GetVolumeId(), targetPath)
 		return &csi.NodePublishVolumeResponse{}, nil
 	}
-	err = n.dirMaker.Make(targetPath, 0750)
-	// MkdirAll returns nil if path already exists
-	if err != nil {
-		return nil, err
-	}
 
 	//TODO support mount options
-	req.GetStagingTargetPath()
-	klog.Infof("Staging target path %s", req.GetStagingTargetPath())
-
-	klog.Infof("GetMount() %v", req.VolumeCapability.GetMount())
-	fsType := req.VolumeCapability.GetMount().FsType
-	klog.Infof("Mounting devicePath %s, on targetPath: %s with FS type: %s",
-		device, targetPath, fsType)
-	err = n.fsMounter.Mount(device.Path, targetPath, fsType, []string{})
+	stagingTarget := req.GetStagingTargetPath()
+	klog.Infof("Staging target path %s", stagingTarget)
+
+	fsType := ""
+	if block {
+		// Make sure target file exists for bind mount
+		f, err := os.OpenFile(targetPath, os.O_CREATE, os.FileMode(0644))
```

Comment:
Why is this necessary?  Will it not get overridden/hidden when bind mount occurs?

Reply:
also, Is there a reason why opening this file with 0644 is important? or would something like what we have below be fine?

Either way, it might be nice to have a helper function `fileExists` here to reduce some of the branching logic in this function. 

```
func fileExists(path string) (bool, error) {
        _, err := os.Stat(path)
        exists := false

        if err == nil {
                exists = true
        } else if errors.Is(err, os.ErrNotExist) {
                err = nil
        }
        return exists, err
}
```

Reply:
Right but I have to create a file so I can bind mount on top of it. If the file doesn't exist, it will give a message about the mount point not existing. I am bind mounting a single file, not a filesystem.

Reply:
huh, so essentially creating a dummy file that gets shadowed by the mount, weird

Reply:
I can probably do 640 to match the 750 for the directory in case of filesystem volume mode. Let me see if that works.